src/BeerBrew/BeerBrew.ino

```diff
@@ -7,462 +7,509 @@
+            break;
+        case STAGE_MASHING:
+            if (sub_stage == 0) {
+                stage1_temp = loop_value(stage1_temp, (1 * plus_minus), 0, 100);
+            } else if (sub_stage == 1) {
+                stage1_time_h = loop_value(stage1_time_h, (1 * plus_minus), 0, 99);
+            } else if (sub_stage == 2) {
+                stage1_time_m = loop_value(stage1_time_m, (1 * plus_minus), 0, 59);
+            }
+            break;
+        case STAGE_MASHING_OUT:
+            stage2_temp = loop_value(stage2_temp, (1 * plus_minus), 0, 100);
+            break;
+        case STAGE_BOIL:
+            if (sub_stage == 0) {
+                stage3_time_h = loop_value(stage3_time_h, (1 * plus_minus), 0, 99);
+            } else if (sub_stage == 1) {
+                stage3_time_m = loop_value(stage3_time_m, (15 * plus_minus), 0, 59);
+            }
+            break;
+        case STAGE_FERMERTATION:
+            stage4_temp = loop_value(stage4_temp, (1 * plus_minus), 0, 100);
+            break;
+    }
+}
+
+void display_current_temperature(float current_temp, int target_temp) {
+    char lcdtemp[7] = "";
+    char line[17];
+    format_temperature(lcdtemp, current_temp);
+    sprintf(line, "s:%d %sC->%3dC", stage, lcdtemp, target_temp);
+    lcd_print_line1(line);
+}
+
+void display_current_time(char* current_time) {
+    char line[17];
+    sprintf(line, "Time:   %s", current_time);
+    lcd_print_line2(line);
+}
+
+bool check_temperature(float tempvalue, float templimit, float overflow, bool greater_than) {
+    if (greater_than) {
+        // FIXME: over 100C water temperature overflow?
+        templimit = relay ? templimit + overflow : templimit - overflow;
+        return tempvalue > templimit;
+    } else {
+        templimit = relay ? templimit - overflow : templimit + overflow;
+        return tempvalue < templimit;
+    }
+}
+
+bool check_time(int pass_time, int time_hour, int time_min) {
+    int set_time = 0;
+    set_time = time_hour * 3600 + time_min * 60;
+    return set_time - pass_time <= 0;
+}
+
+void get_display_time(tmElements_t start_t, tmElements_t end_t, int* pass_time, char* lcdtime) {
+    int pass_hour;
+    int pass_minute;
+    int pass_second;
+    char *pTime = lcdtime;
+    
+    *pass_time = (end_t.Hour - start_t.Hour) * 3600 + (end_t.Minute - start_t.Minute) * 60 + (end_t.Second - start_t.Second);
+    pass_hour = floor(*pass_time / 3600);
+    pass_minute = floor((*pass_time - (pass_hour * 3600)) / 60);
+    pass_second = *pass_time - pass_hour * 3600 - pass_minute * 60;
+    pTime = build_time_str(pTime, pass_hour, 2, ':');
+    pTime = build_time_str(pTime, pass_minute , 2, ':');
+    pTime = build_time_str(pTime, pass_second, 2, ' ');
+#ifdef DEBUG
+    char log[64] = "";
+    sprintf(log, "Time: %02d:%02d:%02d\tpassed: %d", end_t.Hour, end_t.Minute, end_t.Second, *pass_time);
     Serial.println(log);
+#endif
 }
 
+void work() {
+    float tempvalue;
+    char lcdtime[17] = "";
```

Comment:
it will leave a ')' on the right side of the time
we could add a space between current time and left time